### ISSUE TITLE
Wildcard request rewriting 

### ISSUE THREAD

**Issue by pavelnikolov** [projectcontour/contour#4364]
**Please describe the problem you have**

**tl;dr** I am trying to rewrite my path from `/prometheus/account/*/api/v1/query` to `/admin/prometheus/api/v1/query`.

I am using Contour as ingress controller in my Kubernetes cluster. I have a multi-tenanted Prometheus and I want to allow my users to access three paths `/api/v1/query`, `/api/v1/query_range` and `/federate`.  I am using [PromAuthProxy](https://github.com/cherti/PromAuthProxy) to restrict the returned metrics if the user is authorized to access the metrics for the given account ID.

This is my `HttpProxy` configuration:
```yaml
# inside the root HttpProxy
  ...
  includes:
    ...
    - name: promauthproxy
      namespace: customer-monitoring
      conditions:
        - prefix: /prometheus/account
   ...
```
this is the `promauthproxy` definition:
```yaml
apiVersion: projectcontour.io/v1
kind: HTTPProxy
metadata:
  name: promauthproxy
  namespace: customer-monitoring
spec:
  routes:
    - pathRewritePolicy:
        replacePrefix:
          - prefix: /prometheus/account/*/federate
            replacement: /admin/prometheus/federate
          - prefix: /prometheus/account/*/api/v1/query
            replacement: /admin/prometheus/api/v1/query
          - prefix: /prometheus/account/*/api/v1/query_range
            replacement: /admin/prometheus/api/v1/query_range
      services:
        - name: promauthproxy
          port: 8080
      authPolicy:
        context:
          setPromAuthProxyHeader: "true"
```

I am trying to match the wildcard value, in my case a number, rewrite the path and set the ID in a header if the user passes auth. The above doesn't work. Unfortunately, I am not able to to rewrite a wildcard path.
I've tried also setting a prefix condition like this: `/prometheus/account/*/` but to no avail. Is there any way for a Contour HttpProxy to rewrite a path `/some/account/*/path` to `/admin/path` where the wildcard is just a number (i.e. [0-9]+)?

I also tried changing my `promauthproxy` definition to:
```yaml
apiVersion: projectcontour.io/v1
kind: HTTPProxy
metadata:
  name: promauthproxy
  namespace: customer-monitoring
spec:
  routes:
    - pathRewritePolicy:
        replacePrefix:
          - replacement: /admin/prometheus
      services:
        - name: promauthproxy
          port: 8080
      authPolicy:
        context:
          setPromAuthProxyHeader: "true"
```
But unfortunately the above doesn't work either. If I try to access
```
/prometheus/account/5/api/v1/query?query=<some_query>
```
then it gets rewritten to:
```
/admin/prometheus/5/api/v1/query?query=<some_query>
```
This is a showstopper for us and I have to switch back to using Nginx : (
Is there anything I can do in Contour to get rid of the `/5` in the above example?

**Comment by skriss** [comment]
Duplicate of #4241.

**Comment by youngnick** [comment]
@pavelnikolov, I'm going to close this out in favor of discussing on #4241, please comment here if you still want this one open.
